lefthk/src/config.rs

```diff
@@ -175,3 +181,140 @@ fn propagate_exit_chord(chords: Vec<&mut Keybind>, exit_chord: Option<Keybind>)
+    }
+
+    #[test]
+    fn parse_empty_config() {
+        let config = r#"Config(
+    default_modifier: ["Mod4", "Shift"],
+    keybinds: []
+)"#;
+        let conf = Cfg::from_string(config.to_string());
+        assert!(conf.is_ok());
+        let conf = conf.unwrap();
+        assert_eq!(conf.default_modifier.len(), 2);
+        assert_eq!(
+            conf.default_modifier,
+            vec!["Mod4".to_string(), "Shift".to_string()]
+        );
+        let conf_mapped = conf.mapped_bindings();
+
+        // Verify implementation
+        assert_eq!(conf_mapped.len(), 0);
+    }
+
+    #[test]
+    fn parse_none_config() {
+        // Define empty string
+        let conf = Cfg::from_string(String::new());
+        assert!(conf.is_err());
+    }
+
+    #[test]
+    fn parse_sub_keybind_config() {
+        let config = r#"#![enable(implicit_some)]
+Config(
+    default_modifier: ["Mod4", "Shift"],
+    keybinds: [
+        Keybind(
+            command: Chord([
+                Keybind(
+                    command: Execute("st -e htop"),
+                    modifier: ["Mod4"],
+                    key: Key("c"),
+                ),
+            ]),
+            modifier: ["Mod4"],
+            key: Key("c"),
+        ),
+        Keybind(
+            command: Chord([
+                Keybind(
+                    command: Execute("st -e htop"),
+                    key: Key("c"),
+                ),
+            ]),
+            key: Key("c"),
+        ),
+    ]
+)"#;
+        let conf = Cfg::from_string(config.to_string());
+        assert!(conf.is_ok());
+        let conf = conf.unwrap();
+        assert_eq!(conf.default_modifier.len(), 2);
+        assert_eq!(
+            conf.default_modifier,
+            vec!["Mod4".to_string(), "Shift".to_string()]
+        );
+        let conf_mapped = conf.mapped_bindings();
+
+        // Verify default modifier implementation
+        let default_keybind = conf_mapped.last().unwrap();
+        assert_eq!(default_keybind.modifier.len(), 2);
+        assert_eq!(default_keybind.modifier, conf.default_modifier);
+        assert_eq!(
+            default_keybind.command,
+            Command::Chord(vec![Keybind {
+                command: Command::Execute("st -e htop".to_string()),
+                modifier: vec!["Mod4".to_string(), "Shift".to_string()],
+                key: "c".to_string(),
+            }])
+        );
+
+        // Verify custom modifier implementation
+        let custom_keybind = conf_mapped.first().unwrap();
+        assert_eq!(custom_keybind.modifier.len(), 1);
+        assert_eq!(custom_keybind.modifier, vec!["Mod4".to_string()]);
+        assert_eq!(
+            custom_keybind.command,
+            Command::Chord(vec![Keybind {
+                command: Command::Execute("st -e htop".to_string()),
+                modifier: vec!["Mod4".to_string()],
+                key: "c".to_string(),
+            }])
+        );
+    }
+}
```

Comment:
Could you move this test to test.rs

Reply:
Sorry for the delay, I just uploaded an update where I moved the tests as you requested